packages/core/src/components/icon/icon.tsx

```diff
@@ -4,110 +4,4 @@
  * Licensed under the terms of the LICENSE file distributed with this project.
  */
 
-import classNames from "classnames";
-import * as React from "react";
-
-import { IconName, IconSvgPaths16, IconSvgPaths20 } from "@blueprintjs/icons";
-import { Classes, IIntentProps, IProps } from "../../common";
-
-export { IconName };
-
-export interface IIconProps extends IIntentProps, IProps {
-    /** This component does not support custom children. Use the `icon` prop. */
-    children?: never;
-
-    /**
-     * Color of icon. Equivalent to setting CSS `fill` property.
-     */
-    color?: string;
-
-    /**
-     * Name of a Blueprint UI icon, or an icon element, to render.
-     * This prop is required because it determines the content of the component, but it can
-     * be explicitly set to falsy values to render nothing.
-     *
-     * - If `null` or `undefined` or `false`, this component will render nothing.
-     * - If given an `IconName` (a string literal union of all icon names),
-     *   that icon will be rendered as an `<svg>` with `<path>` tags.
-     * - If given a `JSX.Element`, that element will be rendered and _all other props on this component are ignored._
-     *   This type is supported to simplify usage of this component in other Blueprint components.
-     *   As a consumer, you should never use `<Icon icon={<element />}` directly; simply render `<element />` instead.
-     */
-    icon: IconName | JSX.Element | false | null | undefined;
-
-    /**
-     * Size of the icon, in pixels.
-     * Blueprint contains 16px and 20px SVG icon images,
-     * and chooses the appropriate resolution based on this prop.
-     * @default Icon.SIZE_STANDARD = 16
-     */
-    iconSize?: number;
-
-    /** CSS style properties. */
-    style?: React.CSSProperties;
-
-    /**
-     * Description string.
-     * Browsers usually render this as a tooltip on hover, whereas screen
-     * readers will use it for aural feedback.
-     * By default, this is set to the icon's name for accessibility.
-     */
-    title?: string | false | null;
-}
-
-export class Icon extends React.PureComponent<IIconProps & React.SVGAttributes<SVGElement>> {
```

Comment:
🤔 `Icon` component in `core` could _add_ `intent` support so we don't have to push the intent stuff down one package level? though you've moved it nicely and non-breaking-ly here

Reply:
That would mean that the `intent` prop isn't available in the `<*Icon>` exports from `icons` (it'd have to be wrapped in `<Icon>` for that). Doesn't sound like that big of a deal, what do you think?